src/ts/services/test/MatcherService.spec.ts

```diff
@@ -113,4 +121,53 @@ describe("MatcherService", () => {
       done();
     });
   });
+
+  describe("handling skipRanges", () => {
+    const blockText = "ABCDEF";
+    const outgoingText = "BDF";
+    const blockWithSkipRanges = {
+      id: "id",
+      from: 0,
+      to: 6,
+      text: blockText,
+      skipRanges: [
+        { from: 0, to: 0 },
+        { from: 2, to: 2 },
+        { from: 4, to: 4 }
+      ]
+    };
+    it("should remove the skipped ranges from block text as they're passed to the Typerighter service", done => {
+      const service = createMatcherService();
+      fetchMock.post(`${endpoint}/check`, 400);
+      service.requestFetchMatches();
+
+      store.emit("STORE_EVENT_NEW_MATCHES", requestId, [blockWithSkipRanges]);
+      setTimeout(() => {
+        const { blocks } = getLastRequest();
+        const expectedBlocks = [
+          {
+            id: "id",
+            from: 0,
+            to: 3,
+            text: outgoingText
+          }
+        ];
+        expect(blocks).toEqual(expectedBlocks);
+        done();
+      });
+    });
+
+    it("On receipt of matches, map matches that succeed skipped ranges through those ranges to ensure they're correct", done => {
+      const service = createMatcherService();
+      // const response = createResponse(['F'], 2);
+      fetchMock.post(`${endpoint}/check`, 400);
+      service.requestFetchMatches();
+
+      store.emit("STORE_EVENT_NEW_MATCHES", requestId, [blockWithSkipRanges]);
+      setTimeout(() => {
+        // @todo
```

Comment:
I assume there should be some test logic in here?

Reply:
Added 👍